library_of_babel.py

```diff
@@ -53,42 +53,78 @@ def test():
     assert test_string in getPage(search(test_string))
     print ('Tests completed')
 
+def main_filed(input_array):
```

Comment:
Instead of having a bunch of repeated code, could you combine these two methods, but have any output go to a function which takes in a boolean you set earlier as to whether to write it to file or to stdout?